lib/vrl/compiler/src/function.rs

```diff
@@ -433,11 +433,11 @@ impl ArgumentList {
         self.closure = Some(closure);
     }
 
-    fn optional_expr(&mut self, keyword: &'static str) -> Option<Expr> {
-        self.arguments.remove(keyword)
+    pub(crate) fn optional_expr(&mut self, keyword: &'static str) -> Option<Expr> {
+        self.arguments.get(keyword).cloned()
```

Comment:
This looks like a change in behavior (from `remove` to `get`). Is that intended?

Reply:
Yes, that is intentional. The `remove` is not _required_, it was just a nice way to not have to `clone`. This happens are compile time so the perf impact isn't really important. This was preventing me from checking an argument for 2 different types (string _or_ query) since the first would remove the argument even if it failed, then the 2nd would panic.